lib/Driver/Doctrine/DBAL/DBALDriver.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/**
+ * (c) FSi sp. z o.o. <info@fsi.pl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FSi\Component\DataSource\Driver\Doctrine\DBAL;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use FSi\Component\DataSource\Driver\Doctrine\DBAL\Exception\DBALDriverException;
+use FSi\Component\DataSource\Driver\DriverAbstract;
+
+class DBALDriver extends DriverAbstract
+{
+    /**
+     * Default alias for entity during building query when no alias is specified.
+     */
+    const DEFAULT_ENTITY_ALIAS = 'e';
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $table;
+
+    /**
+     * Alias, that can be used with preconfigured query when fetching one entity and field mappings
+     * don't have mappings prefixed with aliases.
+     *
+     * @var string
+     */
+    private $alias;
+
+    /**
+     * Query builder available during preGetResult event.
+     *
+     * @var QueryBuilder
+     */
+    private $query;
+
+    /**
+     * Query builder available during preGetResult event.
+     *
+     * @var QueryBuilder
+     */
+    private $currentQuery;
+
+    /**
+     * @var string|\Closure
+     */
+    private $indexField;
+
+    public function __construct(
+        array $extensions = array(),
+        Connection $connection,
+        $table,
+        $alias = null,
+        $indexField = null
+    ) {
+        parent::__construct($extensions);
+        $this->connection = $connection;
+        $this->table = $table;
+        $this->indexField = $indexField;
+
+        if (is_string($alias)) {
+            $this->alias = (string) $alias;
+        } else {
+            $this->alias = self::DEFAULT_ENTITY_ALIAS;
+        }
+
+        if ($table instanceof QueryBuilder) {
```

Comment:
similar in [ORM driver](https://github.com/fsi-open/datasource/blob/master/lib/Driver/Doctrine/ORM/DoctrineDriver.php#L82), 2 parameters?